lms/static/js/spec/student_account/helpers.js

```diff
@@ -85,6 +86,92 @@ define(['underscore'], function(_) {
+
+            }else {
+                expect($($element.find('.u-field-value')[0]).text()).toBe(view.modelValue());
+            }
+        } else {
+            throw new Error('Unexpected field type: ' + view.fieldType);
+        }
+    };
+
+    var expectProfilePrivacyFieldTobeRendered = function(learnerProfileView, othersProfile) {
+
+        var accountPrivacyElement = learnerProfileView.$('.wrapper-profile-field-account-privacy');
+        var privacyFieldElement = $(accountPrivacyElement).find('.u-field');
+
+        if (othersProfile) {
+            expect(privacyFieldElement.length).toBe(0);
+        } else {
+            expect(privacyFieldElement.length).toBe(1);
+            expectProfileElementContainsField(privacyFieldElement, learnerProfileView.options.accountPrivacyFieldView)
+        }
+    };
+
+    var expectSectionOneTobeRendered = function(learnerProfileView) {
+
+        var sectionOneFieldElements = $(learnerProfileView.$('.wrapper-profile-section-one')).find('.u-field');
+
+        expect(sectionOneFieldElements.length).toBe(learnerProfileView.options.sectionOneFieldViews.length);
+
+        _.each(sectionOneFieldElements, function (sectionFieldElement, fieldIndex) {
+            expectProfileElementContainsField(sectionFieldElement, learnerProfileView.options.sectionOneFieldViews[fieldIndex]);
+        });
+    };
+
+    var expectSectionTwoTobeRendered = function(learnerProfileView) {
+
+        var sectionTwoElement = learnerProfileView.$('.wrapper-profile-section-two');
+        var sectionTwoFieldElements = $(sectionTwoElement).find('.u-field');
+
+        expect(sectionTwoFieldElements.length).toBe(learnerProfileView.options.sectionTwoFieldViews.length);
+
+         _.each(sectionTwoFieldElements, function (sectionFieldElement, fieldIndex) {
+            expectProfileElementContainsField(sectionFieldElement, learnerProfileView.options.sectionTwoFieldViews[fieldIndex]);
+        });
+    };
+
+    var expectProfileSectionsAndFieldsToBeRendered = function (learnerProfileView, othersProfile) {
+        expectProfilePrivacyFieldTobeRendered(learnerProfileView, othersProfile);
+        expectSectionOneTobeRendered(learnerProfileView);
+        expectSectionTwoTobeRendered(learnerProfileView);
+    };
+
+    var expectLimitedProfileSectionsAndFieldsToBeRendered = function (learnerProfileView, othersProfile) {
+        expectProfilePrivacyFieldTobeRendered(learnerProfileView, othersProfile);
+
+        var sectionOneFieldElements = $(learnerProfileView.$('.wrapper-profile-section-one')).find('.u-field');
+
+        expect(sectionOneFieldElements.length).toBe(1);
+        _.each(sectionOneFieldElements, function (sectionFieldElement, fieldIndex) {
+            expectProfileElementContainsField(sectionFieldElement, learnerProfileView.options.sectionOneFieldViews[fieldIndex]);
+        });
+
+        expect($('.profile-private--message').text()).toBe('This edX learner is not currently sharing their profile details.')
```

Comment:
If you're looking at your own profile: You are currently sharing a limited profile.

If you're looking at someone else's profile: This edX learner is currently sharing a limited profile.